pylint/checkers/base.py

```diff
@@ -2511,14 +2511,18 @@ def _check_callable_comparison(self, node):
         left_operand, right_operand = node.left, node.ops[0][1]
         # this message should be emitted only when there is comparison of bare callable
         # with non bare callable.
-        if (
-            sum(
-                1
-                for operand in (left_operand, right_operand)
-                if isinstance(utils.safe_infer(operand), bare_callables)
-            )
-            == 1
-        ):
+        number_of_bare_callables = 0
+        for operand in left_operand, right_operand:
+            inferred = utils.safe_infer(operand)
+            if isinstance(inferred, bare_callables) and not any(
+                # Ignore callables that raise
+                isinstance(x, nodes.Raise)
+                # Or typing constants
+                or "typing._SpecialForm" in inferred.decoratornames()
+                for x in utils.safe_infer(operand).body
```

Comment:
```suggestion
                for x in inferred.body
```

Can't we safe an inference call here?

Reply:
Yes, that was the whole reason for rewriting this as a for loop. I guess between rebases or something I lost it :D